apps/ledger-live-mobile/src/screens/CustomImage/PreviewPreEdit.tsx

```diff
@@ -0,0 +1,285 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { Button, Flex, InfiniteLoader, Text } from "@ledgerhq/native-ui";
+import { ImagePreviewError } from "@ledgerhq/live-common/customImage/errors";
+import { NativeSyntheticEvent, ImageErrorEventData } from "react-native";
+import { useTranslation } from "react-i18next";
+import {
+  EventListenerCallback,
+  EventMapCore,
+  StackNavigationState,
+  useFocusEffect,
+} from "@react-navigation/native";
+import { StackNavigationEventMap } from "@react-navigation/stack";
+import { SafeAreaView } from "react-native-safe-area-context";
+
+import {
+  BaseComposite,
+  StackNavigatorProps,
+} from "../../components/RootNavigator/types/helpers";
+import { CustomImageNavigatorParamList } from "../../components/RootNavigator/types/CustomImageNavigator";
+import { NavigatorName, ScreenName } from "../../const";
+import {
+  downloadImageToFile,
+  importImageFromPhoneGallery,
+} from "../../components/CustomImage/imageUtils";
+import { ImageFileUri } from "../../components/CustomImage/types";
+import { targetDimensions } from "./shared";
+import FramedImage from "../../components/CustomImage/FramedImage";
+import ImageProcessor, {
+  Props as ImageProcessorProps,
+  ProcessorPreviewResult,
+  ProcessorRawResult,
+} from "../../components/CustomImage/ImageProcessor";
+import useCenteredImage, {
+  Params as ImageCentererParams,
+  CenteredResult,
+} from "../../components/CustomImage/useCenteredImage";
+
+const DEFAULT_CONTRAST = 1;
+
+type NavigationProps = BaseComposite<
+  StackNavigatorProps<
+    CustomImageNavigatorParamList,
+    ScreenName.CustomImagePreviewPreEdit
+  >
+>;
+
+const PreviewPreEdit = ({ navigation, route }: NavigationProps) => {
+  const { t } = useTranslation();
+  const [imageToCrop, setImageToCrop] = useState<ImageFileUri | null>(null);
+  const { params } = route;
+  const { isPictureFromGallery, device } = params;
+
+  const handleError = useCallback(
+    (error: Error) => {
+      console.error(error);
+      navigation.navigate(ScreenName.CustomImageErrorScreen, { error, device });
+    },
+    [navigation, device],
+  );
+
+  /** LOAD SOURCE IMAGE FROM PARAMS */
+  useEffect(() => {
+    let dead = false;
+    if ("imageFileUri" in params) {
+      setImageToCrop({
+        imageFileUri: params.imageFileUri,
+      });
+    } else {
+      const { resultPromise, cancel } = downloadImageToFile(params);
+      resultPromise
+        .then(res => {
+          if (!dead) setImageToCrop(res);
+        })
+        .catch(e => {
+          if (!dead) handleError(e);
+        });
+      return () => {
+        dead = true;
+        cancel();
+      };
+    }
+    return () => {
+      dead = true;
+    };
+  }, [params, setImageToCrop, handleError]);
+
+  const [resizedImage, setResizedImage] = useState<CenteredResult | null>(null);
```

Comment:
maybe rename to `croppedImage` or something like that, and uniformize with handleResizedResult etc.

Reply:
Done